src/libraries/System.Private.CoreLib/src/System/Buffers/Text/Base64Decoder.cs

```diff
@@ -616,6 +627,82 @@ private static OperationStatus DecodeWithWhiteSpaceFromUtf8InPlace(Span<byte> ut
+                0x80808080, 0x80808080, 0x80808080, 0x80808080,
+                0x80808080, 0x80808080, 0x3e808080, 0x3f808080,
+                0x37363534, 0x3b3a3938, 0x80803d3c, 0x80808080).AsSByte();
+            Vector512<sbyte> vbmiLookup1 = Vector512.Create(
+                0x02010080, 0x06050403, 0x0a090807, 0x0e0d0c0b,
+                0x1211100f, 0x16151413, 0x80191817, 0x80808080,
+                0x1c1b1a80, 0x201f1e1d, 0x24232221, 0x28272625,
+                0x2c2b2a29, 0x302f2e2d, 0x80333231, 0x80808080).AsSByte();
+            Vector512<byte> vbmiPackedLanesControl = Vector512.Create(
+                0x06000102, 0x090a0405, 0x0c0d0e08, 0x16101112,
+                0x191a1415, 0x1c1d1e18, 0x26202122, 0x292a2425,
+                0x2c2d2e28, 0x36303132, 0x393a3435, 0x3c3d3e38,
+                0x00000000, 0x00000000, 0x00000000, 0x00000000).AsByte();
+
+            Vector512<sbyte> mergeConstant0 = Vector512.Create(0x01400140).AsSByte();
+            Vector512<short> mergeConstant1 = Vector512.Create(0x00011000).AsInt16();
+
+            // This algorithm requires AVX512VBMI support.
+            // Vbmi was first introduced in CannonLake and is avaialable from IceLake on.
+            // This makes it okay to use Vbmi instructions since Vector512.IsHardwareAccelerated returns True only from IceLake on.
+            do
+            {
+                AssertRead<Vector512<sbyte>>(src, srcStart, sourceLength);
+                Vector512<sbyte> str = Vector512.Load(src).AsSByte();
+
+                // Step 1: Translate encoded Base64 input to their original indices
+                // This step also checks for invalid inputs and exits.
+                // After this, we have indices which are verified to have upper 2 bits set to 0 in each byte.
+                // origIndex      = [...|00dddddd|00cccccc|00bbbbbb|00aaaaaa]
+                Vector512<sbyte> origIndex;
+
+                origIndex = Avx512Vbmi.PermuteVar64x8x2(vbmiLookup0, str, vbmiLookup1);
+                Vector512<sbyte> errorVec = (origIndex.AsInt32() | str.AsInt32()).AsSByte();
+                if (errorVec.ExtractMostSignificantBits() != 0)
+                {
+                    break;
+                }
+
+                // Step 2: Now we need to reshuffle bits to remove the 0 bits.
+                // multiAdd1: [...|0000cccc|ccdddddd|0000aaaa|aabbbbbb]
+                Vector512<short> multiAdd1 = Avx512BW.MultiplyAddAdjacent(origIndex.AsByte(), mergeConstant0);
+                // multiAdd1: [...|00000000|aaaaaabb|bbbbcccc|ccdddddd]
+                Vector512<int> multiAdd2 = Avx512BW.MultiplyAddAdjacent(multiAdd1, mergeConstant1);
+
+                // Step 3: Pack 48 bytes
+                str = Avx512Vbmi.PermuteVar64x8(multiAdd2.AsByte(), vbmiPackedLanesControl).AsSByte();
+
+                AssertWrite<Vector512<sbyte>>(dest, destStart, destLength);
+                Vector512.Store(str.AsByte(), dest);
```

Comment:
nit: Most other places in the JIT we do `str.Store(dest)` since its an extension method and can be accessed using instance syntax.

Reply:
> nit: Most other places in the JIT we do `str.Store(dest)` since its an extension method and can be accessed using instance syntax.

I might need to understand how isimdVector works a little better :)

![image](https://github.com/dotnet/runtime/assets/17969209/c0bcbc29-a622-4464-86c7-1d7f750e3b95)

Reply:
I'm not sure I fully understand how this works

![image](https://github.com/dotnet/runtime/assets/17969209/15e20faa-dadc-40c9-8386-9b8d30dddf89)

Reply:
It's likely conflicting because `dest` is a `byte*` while `str` is a `Vector512<sbyte>` and so it can't resolve

`str.Store((sbyte*)dest)` should fix it, or `str.AsByte().Store(dest)`. The former is less IL, most notably.

Reply:
Ah..I messed up and was using `str.AsSbyte().Store(dest)` It's fixed now. Thank you